worker/provisioner/provisioner_test.go

```diff
@@ -1567,19 +1570,28 @@ func assertAvailabilityZoneMachinesDistribution(c *gc.C, obtained []*provisioner
 	c.Assert(max-min, jc.LessThan, 2)
 }
 
+// assertAvailabilityZoneMachinesDistribution checks to see if
+// the distribution groups have been honored.
 func checkAvailabilityZoneMachinesDistributionGroups(c *gc.C, groups map[names.MachineTag][]string, obtained []*provisioner.AvailabilityZoneMachine) error {
 	// Machines in a distribution group should not be in the same
 	// AZ, unless there are more machines in the group, than AZs.
+	obtainedZoneCount := len(obtained)
 	for tag, group := range groups {
+		maxMachineInZoneCount := 1
+		machinesInDGCount := len(group) + 1
```

Comment:
why +1? having trouble understanding this function

Reply:
+1 to include the current machine in calculations.  To be distributed, the number of machines with the same application in 1 AZ should be the number of machines divided by the number of AZ or 1 less to account for the remainder in the division.  Added additional comments.